### PR TITLE
Main

### PR DESCRIPTION
**Pull request by enxebre**
cvo-integration branch is where development of the mao implementation is happening for integrating with CVO and be part of the install/update payload.
pkg/Operator is taken from https://github.com/openshift/machine-config-operator
see https://jira.coreos.com/browse/CLOUD-123
Before merging into master, We'll need to:
- Re vendor Go through go dep nightmare
- Ensure build/image
- Add reconciling for the cluster object, analogous to the machineSet
- Add install folder to be part of the payload
- Test against CVO

### CODE REVIEW

File: pkg/operator/sync.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+const (
+	deploymentRolloutPollInterval = time.Second
+	deploymentRolloutTimeout      = 5 * time.Minute
+
+	customResourceReadyInterval = time.Second
+	customResourceReadyTimeout  = 5 * time.Minute
+)
+
+func (optr *Operator) waitForCustomResourceDefinition(resource *apiextv1beta1.CustomResourceDefinition) error {
```
Comment (vikaschoudhary16) [inline pkg/operator/sync.go]: Why is this needed to check for all the CRDs? Is it just a placeholder and checks will be added to skip non cluster-api CRDs?

**Comment by ingvagabund** [comment]
> add main package

@enxebre can you more elaborate on what it purpose of the PR? Where did you get the `pkg/operator` code? Are there any unit-tests for it?

**Comment by enxebre** [comment]
@ingvagabund cvo-integration branch is where development of the mao implementation is happening for integrating with CVO and be part of the install/update payload.
pkg/Operator is taken from https://github.com/openshift/machine-config-operator
see https://jira.coreos.com/browse/CLOUD-123
Before merging into master, We'll need to:
- Re vendor Go through go dep nightmare
- Ensure build/image
- Add reconciling for the cluster object, analogous to the machineSet
- Add install folder to be part of the payload
- Test against CVO
